backend-tests/tests/test_artifact.py

```diff
@@ -0,0 +1,352 @@
+                deployments.URL_DEPLOYMENTS_ARTIFACTS,
+                files=(
+                    ("description", (None, "description")),
+                    ("size", (None, str(os.path.getsize(artifact)))),
+                    (
+                        "artifact",
+                        (artifact, open(artifact, "rb"), "application/octet-stream"),
+                    ),
+                ),
+            )
+        assert r.status_code == 201
+
+        # create and upload a conflicting mender artifact
+        # conflict because (arm2 / key:value1 / key:value2) are duplicated
+        with self.get_mender_artifact(
+            artifact_name="test",
+            device_types=["arm2", "arm3"],
+            depends=("key1:value1", "key2:value2"),
+            provides=("key3:value3", "key4:value4", "key5:value5"),
+        ) as artifact:
+            r = api_client.with_auth(auth_token).call(
+                "POST",
+                deployments.URL_DEPLOYMENTS_ARTIFACTS,
+                files=(
+                    ("description", (None, "description")),
+                    ("size", (None, str(os.path.getsize(artifact)))),
+                    (
+                        "artifact",
+                        (artifact, open(artifact, "rb"), "application/octet-stream"),
+                    ),
+                ),
+            )
+        assert r.status_code != 201
+
+        # create and upload a non-conflicting mender artifact
+        with self.get_mender_artifact(
+            artifact_name="test",
+            device_types=["arm4"],
+            depends=("key1:value1", "key2:value2"),
+            provides=("key3:value3", "key4:value4", "key5:value5"),
+        ) as artifact:
+            r = api_client.with_auth(auth_token).call(
+                "POST",
+                deployments.URL_DEPLOYMENTS_ARTIFACTS,
+                files=(
+                    ("description", (None, "description")),
+                    ("size", (None, str(os.path.getsize(artifact)))),
+                    (
+                        "artifact",
+                        (artifact, open(artifact, "rb"), "application/octet-stream"),
+                    ),
+                ),
+            )
+        assert r.status_code == 201
+
+    def setup_upload_artifact_selection(self, checksums=("checksum",)):
+        tenant, username, password = self.get_tenant_username_and_password()
+        auth_token = self.get_auth_token(username, password)
+
+        api_client = ApiClient(deployments.URL_MGMT)
+        api_client.headers = {}  # avoid default Content-Type: application/json
+        api_client.with_auth(auth_token)
+
+        # create and upload the mender artifact
+        for checksum in checksums:
+            with self.get_mender_artifact(
+                artifact_name="test",
+                device_types=["arm1"],
+                depends=("rootfs_image_checksum:%s" % checksum,),
+                provides=("rootfs_image_checksum:provided",),
```

Comment:
I might be wrong, but I think `mender-artfiact` also provides this value with the actual checksum by default.

Reply:
But if the following tests passes, then perhaps this value gets overwritten by this flag.

Reply:
Yep, the tests pass, let's keep it as it is

Reply:
It only does when you use `write rootfs-image`, not when using `write module-image`. So for test purposes, the code is correct.